### ISSUE TITLE
DigitalOcean infinite apply loop (forces new resource)

### ISSUE THREAD

**Issue by MartinsThiago** [hashicorp/terraform#12255]
Hi there,

### Terraform Version
v0.8.7

### Affected Resource(s)
- digitalocean_droplet

### Terraform Configuration Files
```hcl
resource digitalocean_droplet stage_server {
  image = "22995881" //ubuntu 16.04 with docker 1.13.1
  name = "test"
  region = "nyc1"
  size = "512mb"
}
```

### Expected Behavior
After a successful `terraform apply`, running `terraform plan` should show
```
Plan: 0 to add, 0 to change, 0 to destroy.
```

### Actual Behavior
It shows
```
Plan: 1 to add, 0 to change, 1 to destroy.
```

### Steps to Reproduce
1. `terraform apply`
2. `terraform plan`

### Possible solution
I think I found where it can be fixed ([See code](https://github.com/hashicorp/terraform/blob/master/builtin/providers/digitalocean/resource_digitalocean_droplet.go#L253-L257))
What happens is that digitalocean_droplet.**image** accepts an **id** or **slug** ([See in docs](https://www.terraform.io/docs/providers/do/r/droplet.html#image))... the specific image I am using offers both options

As the code a mentioned above prefers **slug** over **id**, terraform saves the state container image "ubuntu-16.04"... when running `terraform plan` it displays

![image](https://cloud.githubusercontent.com/assets/8419829/23336300/59291d18-fbaa-11e6-8263-f5d2a3d57388.png)

Despite of just have created the resource, terraform thinks it needs to be updated

**PS: Simply inverting the if code, to prefer ids over slugs won't solve the problem**

**Comment by snowsky** [comment]
I reproduced this issue. I think you are right for the solution. +1

**Comment by snowsky** [comment]
I think this is by design. It will always force to recreate this resource, [here](https://github.com/hashicorp/terraform/blob/master/builtin/providers/digitalocean/resource_digitalocean_droplet.go#L29).

However, we can still use computed (true) instead of ForceNew(true) to let terraform to decide if update when apply.

**Comment by roidelapluie** [comment]
I have proposed a fix in #13879
